local graph isomorphisms do not necessarily map edges onto edges

avekens opened the pull request:
After I had proven that "if two graphs are locally isomorphic, then there is a triangle in one graph if there is a triangle in the other" (see ~grlimgrtri), I wondered if the triangle in the first graph is actually mapped onto a triangle in the other graph by a local isomorphism, as the (formerly) unprecise comment of ~grlimgrtri suggested.  So I tried to prove a theorem "grlimgrtri2" like ~grimgrtri saying that ` (F " T ) ` is a triangle if ` T ` is a triangle and ` F ` a local isomorphism instead of an ordinary isomorphism.  

I ended up in a situation in which I had to prove that an edge ` E ` must be mapped onto an edge ` (F " E ) ` within the 
other graph (which holds for ordinary isomorphisms, see ~ grimedgi).
This, however, turned out to be not generally valid, which can be shown easily looking at the two locally isomorphic generalized Petersen graphs G(5,1) and G(5,2), see ~grlimedgnedg. 

The most I could prove is that a local isomorphism ` F ` between simple pseudographs maps an edge having endpoint ` V` onto an edge having endpoint ``( F ` V)``, see ~grlimgredgex.
 
So I was not able until now to prove a theorem ~grlimgrtri2

``( F e. ( G GraphLocIso H ) ) /\ T e. ( GrTriangles ` G ) ) -> ( F " T ) e. ( GrTriangles ` H ) )``

analogous to ~grimgrtri. There is, however, the possibiliy that "grlimgrtri2" is simply wrong (in general),
so no proof can exist of course. Maybe somebody else has an idea how such a theorem can be proved (or disproved).

Details (in AV's mathbox only):
* theorems ~clnbupgreli, ~clnbgrvtxedg about closed neighborhoods added
* comment of ~grimedg adjusted, new theorem ~grimedgi
* theorems ~grlimedgclnbgr, ~grlimprclnbgr, ~grlimprclnbgredg, ~grlimpredg, ~grlimprclnbgrvtx , ~grlimgredgex about corresponding edges in locally isomorhpic graphs.
* proof of  ~grlimgrtrilem1 shortened
* comment of ~grlimgrtri adjusted
* theorem ~clnbgr3stgrgrlim (analogous to ~clnbgr3stgrgrlic) added
* theorem ~gpg5grlim (isomorphism between the two generalized Petersen graphs G(5,K) of order 10) added
* theorem ~gpg5edgnedg (edges between two corresponding vertices in the two generalized Petersen graphs G(5,K) of order 10) added
* ~grlimedgnedg  added (example for a local graph isomorphism which does not map edges onto edges

savask commented on set.mm:
```diff
@@ -825417,27 +825405,217 @@ Such two graphs are the two generalized Petersen graphs G(5,K) of order 10
+    $d B f g $.  $d B x $.  $d V f g $.  $d W f g $.
+    $( For two locally isormorphic graphs ` G ` and ` H ` and a vertex ` A ` of
+       ` G ` there are two bijections ` f ` and ` g ` mapping the closed
+       neighgborhood ` N ` of ` A ` onto the closed neighborhood ` M ` of
```
"neighborhood"

savask commented on set.mm:
```diff
@@ -825417,27 +825405,217 @@ Such two graphs are the two generalized Petersen graphs G(5,K) of order 10
+    grlimgredgex.g $e |- ( ph -> G e. USPGraph ) $.
+    grlimgredgex.h $e |- ( ph -> H e. USPGraph ) $.
+    grlimgredgex.f $e |- ( ph -> F e. ( G GraphLocIso H ) ) $.
+    $( Local isomorphisms between simple pseudographs map edges onto edges.
```
This comment contradicts the PR title "Local graph isomorphisms do not necessarily map edges onto edges". It seems that `grlimgredgex` proves that an endpoint of an edge in G is mapped to an endpoint of some edge in H.

savask reviewed (approved):
I have only minor remarks.

As for a triangle being always mapped to a triangle, this doesn't hold. You can prove a more general version of your `clnbgr3stgrgrlim`, namely, if all neighborhoods in a graph are isomorphic to each other, then any bijection from the graph to itself is a local isomorphism (of a graph with itself). Now any such graph which has a triangle but doesn't entirely consist of triangles (for instance, a 3-prism) will be a desired counterexample.

avekens commented:
> I have only minor remarks.

I corrected the typos and adjusted the comment. Thank you for your observations.

> As for a triangle being always mapped to a triangle, this doesn't hold. 

> You can prove a more general version of your `clnbgr3stgrgrlim`, namely, if all neighborhoods in a graph are isomorphic to each other, then any bijection from the graph to itself is a local isomorphism (of a graph with itself). 

That is interesting! So I'll try to prove this formally:

```
|- ( ( ( G e. USGraph /\ F : V -1-1-onto-> V )
            /\ A. x e. V A. y e. V ( G ISubGr ( G ClNeighbVtx x ) ) ~=gr ( G ISubGr ( G ClNeighbVtx y ) ) )
                            -> F e. ( G GraphLocIso G ) ) $=
```

> Now any such graph which has a triangle but doesn't entirely consist of triangles (for instance, a 3-prism) will be a desired counterexample.

In the 3-prism, the closed neigborhoods of every vertex are isomorphic to a triangle with an additional edge at one corner. Therefore, every bijection between its vertices is a local isomorphism (by the theorem to be proved). But the bijection which just swaps the vertices (0,0) and (1,0) (adjacent inside and outside vertices) maps the former triangle (0,0),(0,1),(0,2) to (1,0),(0,1),(0,2), which is not a triangle (neither (1,0) and (0,1) nor(1,0) and (0,2) are connected). This should be easy to be proved formally, I hope.

I will do both with my next PR.
